packages/core/crates/core-protocol/src/ticket_aggregation/processor.rs

```diff
@@ -152,29 +152,44 @@ impl AggregationList {
                     .prepare_aggregatable_tickets(&channel_id, epoch, index_start, index_end)
                     .await?
             }
-            AggregationList::TicketList(list) => {
-                if list.is_empty() {
-                    debug!("got empty list of tickets to aggregate");
-                    return Err(ProtocolTicketAggregation("no tickets to aggregate".into()));
-                }
+            AggregationList::TicketList(list) => list,
+        };
 
-                let signer = list[0].signer;
-                let channel_id = list[0].ticket.channel_id;
-
-                if !list.iter().all(|tkt| {
-                    tkt.signer == signer && tkt.ticket.channel_id == channel_id && tkt.status.is_being_aggregated()
-                }) {
-                    error!(
-                        "some tickets to aggregate in the given list do not come from the same channel or are not marked as BeingAggregated"
-                    );
-                    return Err(ProtocolTicketAggregation(
-                        "invalid list of tickets to aggregate given".into(),
-                    ));
-                }
+        if list.is_empty() {
+            debug!("got empty list of tickets to aggregate");
+            return Err(ProtocolTicketAggregation("no tickets to aggregate".into()));
+        }
 
-                list
+        let signer = list[0].signer;
+        let channel_id = list[0].ticket.channel_id;
+
+        let channel_balance = db
+            .read()
+            .await
+            .get_channel(&channel_id)
+            .await?
+            .ok_or(ProtocolTicketAggregation(format!(
+                "channel {channel_id} does not exist"
+            )))?
+            .balance;
+
+        let mut total_amount = Balance::zero(BalanceType::HOPR);
+
+        for tkt in list.iter() {
+            if tkt.signer != signer || tkt.ticket.channel_id != channel_id || !tkt.status.is_being_aggregated() {
+                error!("{tkt} does not belong to the aggregation list");
+                return Err(ProtocolTicketAggregation(
+                    "invalid list of tickets to aggregate given".into(),
+                ));
             }
-        })
+
+            total_amount = total_amount.add(&tkt.ticket.amount);
+            if total_amount.gt(&channel_balance) {
+                return Err(ProtocolTicketAggregation(format!("aggregation list has total value of {total_amount} which is greater than balance {channel_balance} in channel {channel_id}")));
+            }
```

Comment:
Why check here the cumulated ticket balance? Later there's comparison on the final balance which takes the "win_prob" into account, which is more accurate.

Reply:
`Later there's comparison on the final balance...` - where is this exactly ?

Reply:
`final_value`, particularly at https://github.com/hoprnet/hoprnet/blob/3bd25791111fccb855ef77bb87f87dfd2c85c15d/packages/core/crates/core-protocol/src/ticket_aggregation/processor.rs#L324-L327

Reply:
There are two checks, one for each side: at the aggregation requestor (L186) and at the aggregator (L325).